gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagActionReminderScheduler.java

```diff
@@ -124,15 +128,19 @@ public static String createDagActionReminderKey(DagActionStore.DagAction dagActi
    * Creates a JobKey object for the reminder job where the name is the DagActionReminderKey from above and the group is
    * the flowGroup
    */
-  public static JobKey createJobKey(DagActionStore.DagAction dagAction) {
-    return new JobKey(createDagActionReminderKey(dagAction), dagAction.getFlowGroup());
+  public static JobKey createJobKey(DagActionStore.DagAction dagAction, boolean deadlineReminder) {
+    return new JobKey(createDagActionReminderKey(dagAction), deadlineReminder ? DeadlineReminderKeyGroup : RetryReminderKeyGroup);
+  }
+
+  private static TriggerKey createTriggerKey(DagActionStore.DagAction dagAction, boolean deadlineReminder) {
+    return new TriggerKey(createDagActionReminderKey(dagAction), deadlineReminder ? DeadlineReminderKeyGroup : RetryReminderKeyGroup);
   }
 
   /**
    * Creates a jobDetail containing flow and job identifying information in the jobDataMap, uniquely identified
    *  by a key comprised of the dagAction's fields.
    */
-  public static JobDetail createReminderJobDetail(DagActionStore.DagActionLeaseObject dagActionLeaseObject) {
+  public static JobDetail createReminderJobDetail(DagActionStore.DagActionLeaseObject dagActionLeaseObject, boolean deadlineReminder) {
```

Comment:
can you update javadoc to explain what this boolean means? is it used for the first time we set reminder for deadline trigger only? all of the other executions are truly "reminders" so the terminology is too ambiguous here

Reply:
done. yes. going forward lets call "truly reminders" as "retry reminders"